src/Google.Bigquery.V2/BigqueryClient.TableCrud.cs

```diff
@@ -0,0 +1,186 @@
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Attempts to fetch a table specified by project ID, dataset ID and table ID, creating it if it doesn't exist.
+        /// This method just creates a <see cref="TableReference"/> and delegates to <see cref="GetOrCreateTable(TableReference, TableSchema)"/>.
+        /// </summary>
+        /// <param name="projectId">The project ID. Must not be null.</param>
+        /// <param name="datasetId">The dataset ID. Must not be null.</param>
+        /// <param name="tableId">The table ID. Must not be null.</param>
+        /// <param name="schema">The schema to use to create the table if necessary. Must not be null.</param>
+        /// <returns>The existing or new table.</returns>
+        public virtual BigqueryTable GetOrCreateTable(string projectId, string datasetId, string tableId, TableSchema schema)
+            => GetOrCreateTable(GetTableReference(projectId, datasetId, tableId), schema);
+
+        /// <summary>
+        /// Attempts to fetch the specified table within this client's project, creating it if it doesn't exist.
+        /// This method just creates a <see cref="TableReference"/> and delegates to <see cref="GetOrCreateTable(TableReference, TableSchema)"/>.
+        /// </summary>
+        /// <param name="datasetId">The dataset ID. Must not be null.</param>
+        /// <param name="tableId">The table ID. Must not be null.</param>
+        /// <param name="schema">The schema to use to create the table if necessary. Must not be null.</param>
+        /// <returns>The existing or new table.</returns>
+        public virtual BigqueryTable GetOrCreateTable(string datasetId, string tableId, TableSchema schema)
+            => GetOrCreateTable(GetTableReference(datasetId, tableId), schema);
+
+        /// <summary>
+        /// Attempts to fetch a table, creating it if it doesn't exist.
+        /// </summary>
+        /// <param name="tableReference">A fully-qualified identifier for the table. Must not be null.</param>
+        /// <param name="schema">The schema to use to create the table if necessary. Must not be null.</param>
+        /// <returns>The existing or new table.</returns>
+        public virtual BigqueryTable GetOrCreateTable(TableReference tableReference, TableSchema schema)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Deletes a table specified by project ID, dataset ID and table ID.
+        /// This method just creates a <see cref="TableReference"/> and delegates to <see cref="DeleteTable(TableReference)"/>.
+        /// </summary>
+        /// <param name="projectId">The project ID. Must not be null.</param>
+        /// <param name="datasetId">The dataset ID. Must not be null.</param>
+        /// <param name="tableId">The table ID. Must not be null.</param>
+        public virtual void DeleteTable(string projectId, string datasetId, string tableId)
```

Comment:
Hmm, should all methods that take a "string projectId" also have an overload to accept a ProjectReference...
TooManyOverloadsException?